From f86a02f31783acb1be29daf748f8385e8a66f113 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Wed, 11 Apr 2012 23:37:19 -0400
Subject: [PATCH] Make list of gdk targets non-automagic

Otherwise, it is difficult e.g. to build a wayland-only gtkmm on a
build machine that has gdk with both x11 and wayland targets installed.
The argument to --with-gdk-targets is comma-delimeted to make quoting
easier in automated build scripts.

https://bugzilla.gnome.org/show_bug.cgi?id=673963
---
 configure.ac |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5c85b7e..141f39a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -52,7 +52,13 @@ gdk_target_quartz=no
 gdk_target_x11=no
 gdk_target_wayland=no
 gdk_target_broadway=no
-for target in `$PKG_CONFIG --variable=targets gtk+-3.0`; do
+AC_ARG_WITH(gdk_targets, AS_HELP_STRING([--with-gdk-targets@<:@=TARGETS@:>@], [comma-delimeted list of gdk targets (default is "", meaning auto-detect)]), ,[with_gdk_targets=])
+if test -z "$with_gdk_targets"; then
+    gdk_targets=`$PKG_CONFIG --variable=targets gtk+-3.0`
+else
+    gdk_targets=`echo $with_gdk_targets | tr , ' '`
+fi
+for target in $gdk_targets; do
     case "$target" in 
        win32)
           gdk_target_win32=yes
-- 
1.7.8.5

